app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectsListActivity.kt

```diff
@@ -29,8 +29,20 @@ class ProjectsListActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_projects_list)
+
+        val fromLink = intent.getBooleanExtra("fromLink", false)
+        var projectId = ""
+        if (fromLink) {
+            var temp = intent.getStringExtra("projectId")
+            if (temp == null) {
+                projectId = ""
+            } else {
+                projectId = temp
+            }
```

Comment:
it could be simplified by
```suggestion
            projectId = intent.getStringExtra("projectId") ?: ""
```